tests/code/test_emailHandler.py

```diff
@@ -119,6 +126,65 @@ def test_email_log():
             assert emailLog.sender == User.get_by_id("ramsayb2")
             transaction.rollback()
 
+@pytest.mark.integration
+def test_recipients_category():
+    with app.test_request_context():
+        with mainDB.atomic() as transaction:
+            url_domain = urlparse(request.base_url).netloc
+            raw_form_data = {"templateIdentifier": "Test",
+                "programID":"3",
+                "eventID":"1",
+                "recipientsCategory": "Eligible Students"}
+
+            testSender = User.get_by_id('ramsayb2')
+
+            email = EmailHandler(raw_form_data, url_domain, testSender)
+            email.process_data()
+            assert email.recipients == []
+
+            # Add partont to All Volunteer Training event: NOT banned and IS trained
+            newTrainedStudent = EventParticipant.create(user = "partont", event = 14)
+            email.process_data()
+            assert email.recipients == [User.get_by_id("partont")]
+
+            # Add ayisie to a non-all volunteer training event: NOT banned and NOT trained
+            newTrainedStudent = EventParticipant.create(user = "ayisie", event = 5)
+            email.process_data()
+            assert email.recipients ==  [User.get_by_id("partont")]
+
+            # Train ayisie so they show up in the results: NOT banned and IS trained
+            newTrainedStudent = EventParticipant.create(user = "ayisie", event = 14)
+            email.process_data()
+            assert email.recipients ==  [User.get_by_id("partont"),User.get_by_id("ayisie")]
+            newTrainedStudent.delete_instance()
+
+            # Add khatts to All Volunteer Training event: IS banned and IS trained
+            newTrainedStudent = EventParticipant.create(user = "khatts", event = 14)
+            email.process_data()
+            assert email.recipients == [User.get_by_id("partont")]
+            newTrainedStudent.delete_instance()
+
+            # Unban khatts while they have All Volunteer Training: NOT banned IS trained
+            ProgramBan.update(endDate = parser.parser("2022-6-23")).where(ProgramBan.user == "khatts").execute()
+            newTrainedStudent = EventParticipant.create(user = "khatts", event = 14)
+            email.process_data()
+            assert email.recipients == [User.get_by_id("partont"), User.get_by_id("khatts")]
+            newTrainedStudent.delete_instance()
+
+            transaction.rollback()

```

Comment:
This rollback should go at the end. It doesn't affect the current tests, but it would be easy for someone to come in and add tests after the one below, not realizing they were missing the rollback